src/components/demos/write/Demo.js

```diff
@@ -15,129 +20,94 @@ const Wrapper = styled.div`
-          ...writeDemo,
-          usernamePasswordPairs: [
-            ...writeDemo.usernamePasswordPairs,
-            usernameInput,
-            passwordInput,
-          ],
-        },
-      })
-      // TODO execute API
-      setDemoOutput('***PARSED API OUTPUT***')
+    if (usernamePasswordPairs.length > 0) {
+      attemptLogin()
     }
-  }
+  }, [usernamePasswordPairs, execute, binaryName, theme])
 
   return (
     <>
       <Header {...props} showClose={true} />
       <Wrapper {...theme.wrapper}>
-        <Box {...writeDemo}>
-          <Container styles={{ height: '100%', paddingTop: '150px' }} size={10}>
-            <form onSubmit={enterUsernameAndPassword}>
-              <Input
-                label={'Username'}
-                theme={writeDemo.theme.form}
-                setInputState={setUsernameInput}
-                upperBound={usernameUpperBound}
-                showInputError={usernameAtMaxLength || noUsernameEntered}
-                InputErrorWarning={UsernameErrorWarning}
-                cySelector={'username'}
-              />
-              <Input
-                label={'Password'}
-                theme={writeDemo.theme.form}
-                setInputState={setPasswordInput}
-                upperBound={passwordUpperBound}
-                inputType={'password'}
-                showInputError={passwordAtMaxLength || noPasswordEntered}
-                InputErrorWarning={PasswordErrorWarning}
-                cySelector={'password'}
-              />
+        {successfulLogin(apiOutput) ? (
+          <>
+            <SecretDesktop {...theme.font} />
+            {!isMorello && <ButtonSide action={switchToMorello} />}
+          </>
+        ) : (
+          <>
+            <Box {...demoState}>
               <Container
+                styles={{ height: '100%', paddingTop: '150px' }}
                 size={10}
-                styles={{
-                  flexDirection: 'column',
-                  alignItems: 'center',
-                  gap: '30px',
-                }}
               >
-                <Button
-                  {...writeDemo.theme.form.loginButton}
-                  data-cy={'login'}
-                  type={'submit'}
-                >
-                  Login
-                </Button>
-                <LoginAttemptText
-                  {...writeDemo.theme.form.loginAttempt}
-                  visibility={demoOutput ? 'visible' : 'hidden'}
-                  data-cy={'login-attempt'}
-                >
-                  {demoOutput}
-                </LoginAttemptText>
+                <LoginForm
+                  demoState={demoState}
+                  showSpinner={awaitingApi}
+                  setUsernamePasswordPairs={setUsernamePasswordPairs}
+                  apiOutput={apiOutput}
+                />
               </Container>
-            </form>
-          </Container>
-        </Box>
+            </Box>
+            {isMorello && apiOutput && (
```

Comment:
I think this should only be triggered once the hack has either been successful or when the hack has been prevented by cheri

Reply:
added https://github.com/digicatapult/morello-ui/pull/30/commits/b0724eb19a28836599799fb48f0279eaca01548a